src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -1,23 +1,214 @@
+    }
+
+    /// <summary>
+    /// Add entity to the EntityNameToDataSourceName dictionary.
+    /// </summary>
+    /// <param name="entityName">EntityName</param>
+    /// <param name="dataSourceName">DatasourceName.</param>
+    /// <returns>True indicating success, False indicating failure.</returns>
+    /// <exception cref="DataApiBuilderException">Not found exception if key is not found.</exception>
+    public bool TryAddEntity(string entityName, string dataSourceName)
+    {
+        CheckDataSourceNamePresent(dataSourceName);
+        return _entityNameToDataSourceName.TryAdd(entityName, dataSourceName);
+    }
+
+    /// <summary>
+    /// Updates the EntityNameToDataSourceName dictionary with the new Entity to datasource mapping.
+    /// </summary>
+    /// <param name="entityName">EntityName.</param>
+    /// <param name="dataSourceName">dataSourceName.</param>
+    /// <exception cref="DataApiBuilderException"></exception>
+    public void UpdateEntityNameToDataSourceName(string entityName, string dataSourceName)
+    {
+        CheckDataSourceNamePresent(dataSourceName);
+        CheckEntityNamePresent(entityName);
+        _entityNameToDataSourceName[entityName] = dataSourceName;
+    }
+
+    /// <summary>
+    /// Removes the entity from the EntityNameToDataSourceName dictionary.
+    /// </summary>
+    /// <param name="entityName">Name of Entity</param>
+    /// <exception cref="DataApiBuilderException">Not found exception if key is not found.</exception>
+    public bool RemoveEntity(string entityName)
+    {
+        CheckEntityNamePresent(entityName);
+        return _entityNameToDataSourceName.Remove(entityName);
+    }
+
+    /// <summary>
+    /// Get the default datasource name.
+    /// </summary>
+    /// <returns>default datasourceName.</returns>
+#pragma warning disable CA1024 // Use properties where appropriate. Reason: Do not want datasource serialized and want to keep it private to restrict set;
+    public string GetDefaultDataSourceName()
+#pragma warning restore CA1024 // Use properties where appropriate
+    {
+        return _defaultDataSourceName;
+    }
+
     /// <summary>
     /// Serializes the RuntimeConfig object to JSON for writing to file.
     /// </summary>
     /// <returns></returns>
-    public string ToJson()
+    public string ToJson(JsonSerializerOptions? jsonSerializerOptions = null)
+    {
+        // get default serializer options if none provided.
+        jsonSerializerOptions = jsonSerializerOptions ?? RuntimeConfigLoader.GetSerializationOptions();
+        return JsonSerializer.Serialize(this, jsonSerializerOptions);
+    }
+
+    private void CheckDataSourceNamePresent(string dataSourceName)
+    {
+        if (!_dataSourceNameToDataSource.ContainsKey(dataSourceName))
+        {
+            throw new DataApiBuilderException($"{nameof(dataSourceName)} could not be found within the config", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
```

Comment:
shouldn't we log the actual value of dataSourceName that is not found as well? So this is debuggable?